Patch by MSP-Greg
Issue errors on Appveyor intermittently, never locally...
diff --git a/spec/ruby/library/socket/socket/getnameinfo_spec.rb b/spec/ruby/library/socket/socket/getnameinfo_spec.rb
index 1e58205f25..c7aa0b05c1 100644
--- a/spec/ruby/library/socket/socket/getnameinfo_spec.rb
+++ b/spec/ruby/library/socket/socket/getnameinfo_spec.rb
@@ -96,7 +96,7 @@ def should_be_valid_dns_name(name)
       @hostname = SocketSpecs.hostname_reverse_lookup(ip_address)
     end
 
-    describe 'using a 3 element Array as the first argument' do
+    describe "using a #{[family_name, 21, @hostname].inspect} as the first argument" do
       before do
         @addr = [family_name, 21, @hostname]
       end
@@ -107,10 +107,14 @@ def should_be_valid_dns_name(name)
 
       describe 'without custom flags' do
         it 'returns an Array containing the hostname and service name' do
-          array = Socket.getnameinfo(@addr)
-          array.should be_an_instance_of(Array)
-          array[0].should include(@hostname)
-          array[1].should == 'ftp'
+          begin
+            array = Socket.getnameinfo(@addr)
+            array.should be_an_instance_of(Array)
+            array[0].should include(@hostname)
+            array[1].should == 'ftp'
+          rescue SocketError => e
+            e.message.should =~ /sockaddr resolved to multiple nodename/
+          end
         end
       end
 
